packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1073,18 +1073,28 @@ function getSchemaForStdScalar(
     case "float64":
       return applyIntrinsicDecorators(program, type, {
         type: "number",
-        format: "double"
+        format: "float64"
       });
     case "float32":
       return applyIntrinsicDecorators(program, type, {
         type: "number",
-        format: "float"
+        format: "float32"
       });
     case "float":
       return applyIntrinsicDecorators(program, type, {
         type: "number",
         format: "float"
       });
+    case "decimal":
+      return applyIntrinsicDecorators(program, type, {
+        type: "number",
```

Comment:
the temp workaround to ci failures and detailed fix would be covered https://github.com/Azure/autorest.typescript/issues/2157